### PR TITLE
Add Department Enum and AssignedDepartment Class

### PR DESCRIPTION
**Pull request by wujy28**
# Add Department and AssignedDepartment Classes

# Summary of change
Added the Department enum and AssignedDepartment class to the model so that Patient can store
their assigned department as an attribute in the future.

# Key changes

## Added AssignedDepartment class
1. Added the class as an attribute to Patient
2. Conducted unit testing on methods in the class

## Added Department enumeration
1. Added the enumeration to store accepted values for departments in the hospital
2. Wrote unit tests for the enumeration

## Fix minor checkstyle issues
1. Fix the trailing whitespace issue in the Gender class

### CODE REVIEW

File: src/main/java/seedu/address/model/patient/Department.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package seedu.address.model.patient;
+
+/**
+ * Represents the departments in the hospital.
+ */
+public enum Department {
```
Comment (thienmy0) [inline src/main/java/seedu/address/model/patient/Department.java:6]: I like how you used enums to represent the Departments!

File: src/test/java/seedu/address/model/patient/DepartmentTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        assertFalse(Department.isValidDepartment("Psych1atry")); // Name does not match
+        assertFalse(Department.isValidDepartment(" Plastic Surgery")); // first character is a space
+        assertFalse(Department.isValidDepartment("Anesthesiology")); // wrong spelling (American)
+        assertFalse(Department.isValidDepartment("Burn Unit")); // department does not exist
+
+        // valid department
+        assertTrue(Department.isValidDepartment("-")); // default department
+        assertTrue(Department.isValidDepartment("Palliative Medicine")); // department exists
+        assertTrue(Department.isValidDepartment("emergency department")); // lower case only
+        assertTrue(Department.isValidDepartment("neUroSURGerY")); // weird casing
```
Comment (thienmy0) [inline src/test/java/seedu/address/model/patient/DepartmentTest.java:43]: I like how you implemented the test cases clearly and followed the testing format of other attributes! Perhaps you could add more cases to test more extensively?

File: src/main/java/seedu/address/model/patient/AssignedDepartment.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * Represents a Patient's assigned department in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDepartment(String)}
+ */
+public class AssignedDepartment {
+
+    public static final String MESSAGE_CONSTRAINTS = "Departments can only take certain values, "
+            + "must adhere to British spelling conventions, "
+            + "and should not be blank.\n"
+            + "Please view the user guide for the list of accepted departments.\n"
+            + "Link: https://ay2324s1-cs2103t-t14-2.github.io/tp/UserGuide.html";
```
Comment (thienmy0) [inline src/main/java/seedu/address/model/patient/AssignedDepartment.java:16]: I like how the constraints for Department are specific and clear!

**Review by thienmy0 — APPROVED** [review]
Looks good to me overall! You may take a look at some of the comments given. Good job! 👍

File: src/main/java/seedu/address/model/patient/AssignedDepartment.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            return false;
+        }
+
+        AssignedDepartment otherDepartment = (AssignedDepartment) other;
+        return assignedDepartment.equals(otherDepartment.assignedDepartment);
+    }
+
+    @Override
+    public int hashCode() {
+        return assignedDepartment.hashCode();
```
Comment (longnguyentan) [inline src/main/java/seedu/address/model/patient/AssignedDepartment.java:73]: This line has not been covered by tests.
```
Codecov / codecov/patch
src/main/java/seedu/address/model/patient/AssignedDepartment.java#L73
Added line #L73 was not covered by tests
```

File: src/main/java/seedu/address/model/patient/Department.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                return d;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns true if the given string represents a valid Department.
+     */
+    public static boolean isValidDepartment(String string) {
```
Comment (longnguyentan) [inline src/main/java/seedu/address/model/patient/Department.java:61]: This method also was not covered by tests.
```
Codecov / codecov/patch
src/main/java/seedu/address/model/patient/Department.java#L63
Added line #L63 was not covered by tests
```

File: src/test/java/seedu/address/model/patient/AssignedDepartmentTest.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+    @Test
+    public void constructor_invalidDepartment_throwsIllegalArgumentException() {
+        String invalidDepartment = "";
+        assertThrows(IllegalArgumentException.class, () -> new AssignedDepartment(invalidDepartment));
+
+
+    }
+    @Test
+    void isValidDepartment() {
```
Comment (longnguyentan) [inline src/test/java/seedu/address/model/patient/AssignedDepartmentTest.java:37]: Can consider some other cases like:
- Having space between characters (e.g. `Orthop aedics`)
- Missing suffix `s` (e.g. `Orthop aedic`)

**Review by longnguyentan — COMMENTED** [review]
Overall are good :) Keep it up

**Pull request merged**